tgui/packages/tgui/interfaces/ApcControl.js

```diff
@@ -0,0 +1,335 @@
+import { Box, Section, Button, Table, Icon, Dimmer, Tabs } from '../components';
+import { map, sortBy } from 'common/collections';
+import { flow } from 'common/fp';
+import { toFixed } from 'common/math';
+import { Window, Fragment } from '../layouts';
+import { useBackend, useLocalState } from '../backend';
+import { pureComponentHooks } from 'common/react';
+import { FlexItem } from '../components/Flex';
+
+export const ApcControl = (props, context) => {
+  const { data } = useBackend(context);
+  return (
+    <Window resizable>
+      {data.authenticated === 1
+      && <ApcLoggedIn />}
+      {data.authenticated === 0
+      && <ApcLoggedOut />}
+    </Window>
+  );
+};
+
+const ApcLoggedOut = (props, context) => {
+  const { act, data } = useBackend(context);
+  const text = data.emagged === 1 ? 'Open' : 'Log In';
+  return (
+    <Window.Content>
+      <Button fluid
+        color={data.emagged === 1 ? '' :'good'}
+        tooltip={text}
+        content={text}
+        onClick={() => act('log-in')} />
+    </Window.Content>
+  );
+};
+
+const ApcLoggedIn = (props, context) => {
+  const { act, data } = useBackend(context);
+  const [
+    tabIndex,
+    setTabIndex,
+  ] = useLocalState(context, 'tab-index', 1);
+  return (
+    <Fragment>
+      <Tabs>
+        <Tabs.Tab
+          selected={tabIndex === 1}
+          onClick={() => {
+            setTabIndex(1);
+            act("check-apcs");
+          }}>
+          APC Control Panel
+        </Tabs.Tab>
+        <Tabs.Tab
+          selected={tabIndex === 2}
+          onClick={() => {
+            setTabIndex(2);
+            act("check-logs");
+          }}>
+          Log View Panel
+        </Tabs.Tab>
+      </Tabs>
+      {data.restoring === 1&& (
+        <Dimmer fontSize="32px">
+          <Icon name="cog" spin={1} />
```

Comment:
Spin is a boolean property.

1. Don't use numbers for booleans, this is not BYOND code.
2. Property without an assigned value is considered `true`, like in normal HTML code.

```suggestion
          <Icon name="cog" spin />
```

Reply:
Thank you